api/_lib/script/build.ts

```diff
@@ -71,10 +71,16 @@ async function main() {
     }
 
     const fullDir = join(packagesDir, pkg);
-    const packageJsonRaw = await fs.readFile(
-      join(fullDir, 'package.json'),
-      'utf-8'
-    );
+    const packageJsonRaw = await fs
+      .readFile(join(fullDir, 'package.json'), 'utf-8')
+      .catch(() => null);
+    if (!packageJsonRaw) {
+      // `package.json` might not exist if this directory exists due to the
+      // Vercel deployment's build cache (even though the package has been
+      // deleted). So skip in that case.
+      continue;
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (non-blocking):** ‏can we log when this happens?